Sources/Apodini/Semantic Model Builder/gRPC/GRPCService/GRPCService.swift

```diff
@@ -23,6 +24,77 @@ class GRPCService {
         self.serviceName = name
         self.app = app.vapor.app
     }
+
+    internal func checkContentType(request: Vapor.Request) -> Bool {
+        let contentType = request.headers.first(name: .contentType)
+        switch contentType {
+        case "application/grpc", "application/grpc+proto":
+            return true
+        default:
+            // GRPC theoretically would also allow for other
+            // types of payload formats, e.g. JSON.
+            // We do only support proto payloads at the moment
+            return false
+        }
+    }
+
+    /// Cuts the given data into the individual GRPC messages it represents.
+    ///
+    /// Cite: https://grpc.io/blog/grpc-on-http2/
+    /// "RPCs are in practice plain HTTP/2 streams.
+    /// Messages are associated with RPCs and get sent as HTTP/2 data frames.
+    /// To be more specific, messages are layered on top of data frames.
+    /// A data frame may have many gRPC messages,
+    /// or if a gRPC message is quite large it might span multiple data frames."
+    ///
+    /// - Parameter data: The data to read the GRPC  messages from
+    internal func getMessages(from data: Data) -> [GRPCMessage] {
+        var data = data
+        var messages: [GRPCMessage] = []
+        var hasNext = false
+        repeat {
```

Comment:
Is it maybe possible to express this loop in terms of `data.count`?

Reply:
Yes it is 😉